tests/foreman/cli/test_ansible.py

```diff
@@ -561,3 +566,107 @@ def test_positive_install_ansible_collection(
         assert result['success'] == '1'
         collection_path = client.execute('ls ~/ansible_collections').stdout
         assert 'oasis_roles' in collection_path
+
+    @pytest.mark.tier3
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    @pytest.mark.parametrize('auth_type', ['admin', 'non-admin'])
+    def test_positive_ansible_variables_imported_with_roles(
+        self, request, auth_type, target_sat, module_org, module_ak_with_cv, rhel_contenthost
+    ):
+        """Verify that when Ansible roles are imported, their variables are imported simultaneously
+
+        :id: 107c53e8-5a8a-4291-bbde-fbd66a0bb85e
+
+        :steps:
+            1. Register a content host with satellite
+            2. Create a custom role with variables and import it into satellite
+            3. Assert that the role is imported along with associated variables
+            4. Assign that role to a host and verify the role is assigned to the host
+            5. Run the Ansible role
+
+        :expectedresults: Verify variables associated to role are also imported along with roles
+
+        :verifies: SAT-28198
+        """
+        username = settings.server.admin_username
+        password = settings.server.admin_password
+        if auth_type == 'non-admin':
+            username = gen_string('alpha')
+            user = target_sat.cli_factory.user(
+                {
+                    'admin': False,
+                    'login': username,
+                    'password': password,
+                    'organization-ids': module_org.id,
+                }
+            )
+            target_sat.cli.User.add_role(
```

Comment:
Does this user have other roles as well? The `Ansible Roles Manager` role doesn't grant permission to edit the host.

Reply:
no, it'll have only this role to grant permissions for assigning role to host, and I think edit host permission might not be required here for this scenario

Reply:
How can you "Assign that role to a host and verify the role is assigned to the host" (step 4) without permission to edit the host?

Reply:
I see step 4 is not done in this method, so maybe I'm missing some details here.

Reply:
yes yes, we're not actually verifying if role is assigned to host, but I don't think it'll require any edit host permissions but might require :view_hosts: permission, right? I'll check this locally and confirm

Reply:
I made a mistake, I mentioned that we should verify role assigned on host, but we're not verifying it, I just added changes for that.
For permissions related to assign role to host/hg, I see its not added to "Ansible Roles Manager" role, but not sure how user is getting `edit_hosts` permissions and it is working for the user. How do you suggest we proceed with this PR?

Reply:
Can you assign the user the `Edit hosts` role?

Reply:
@nofaralfasi As discussed, it works without requiring this role as when I manually tested it and it was working fine

Reply:
@Gauravtalreja1 Could it be a bug? Without edit hosts permission, should we allow user to add role to it?

Reply:
@nofaralfasi @shweta83 It turns out this isn't really a bug, upon digging I found that the user is assigned a "Register hosts" role, which has the ability to view and edit hosts
```
# hammer role filters --name "Register hosts"
----|----------------------------|--------|------------|-----------|----------------|----------------------------------------------------
ID  | RESOURCE TYPE              | SEARCH | UNLIMITED? | OVERRIDE? | ROLE           | PERMISSIONS
----|----------------------------|--------|------------|-----------|----------------|----------------------------------------------------
251 | Host                       | none   | yes        | no        | Register hosts | view_hosts, edit_hosts, create_hosts, destroy_hosts
```

Reply:
Thanks for checking.